Content.Shared/Weapons/Ranged/Components/GunComponent.cs

```diff
@@ -143,7 +143,7 @@ public sealed partial class GunComponent : Component
     /// <summary>
     /// Who the gun is being requested to shoot at directly.
     /// </summary>
-    [ViewVariables]
+    [ViewVariables, Access(Other = AccessPermissions.ReadWrite)]
```

Comment:
You should probably add a function to GunSystem instead.

Reply:
Done.